packages/modelviewer.dev/src/docs-and-examples/sidebar.ts

```diff
@@ -175,18 +176,49 @@ function handleHTMLEntry(htmlEntry: IntersectionObserverEntry) {
   }
 }
 
+/*
+ * for page jump its just easier to restart, so deactivate everything, clear
+ * the global view, then only update with whats in view
+ */
+function handlePageJump(entries: IntersectionObserverEntry[]) {
+  globalCurrentView = [];
+  toRemove = '';
+  previouslyActive = '';
+  updateSidebarView('', 'null');
+  for (const htmlEntry of entries) {
+    const id = htmlEntry.target.getAttribute('id')!;
+    const sidebarIds = getSidebarIdsFromId(id);
+    deactivateSidebar(sidebarIds);
+  }
+  let isAtTop = true;
+  for (const htmlEntry of entries) {
+    if (htmlEntry.intersectionRatio > 0) {
+      if (isAtTop) {
+        isAtTop = false;
+        const id = htmlEntry.target.getAttribute('id')!;
+        const sidebarIds = getSidebarIdsFromId(id);
+        updateSidebarView('', sidebarIds.subcategory);
+      }
+      handleHTMLEntry(htmlEntry);
+    }
+  }
+}
+
 
 /*
  * Update the table of contents based on how the page is viewed.
  */
 export function sidebarObserver(docsOrExamples: 'docs'|'examples') {
   if (docsOrExamples === 'docs') {
     const observer = new IntersectionObserver(entries => {
-      for (const htmlEntry of entries) {
-        handleHTMLEntry(htmlEntry);
+      if (!isFirstOpen && entries.length > 2) {
+        handlePageJump(entries);
+      } else {
+        for (const htmlEntry of entries) {
+          handleHTMLEntry(htmlEntry);
+        }
       }
-      // True the first time all of the observers are set.
-      // entries will be every possible entry on the whole page.
+      // First time all of the observers are set entries will be every entry.
```

Comment:
Not quite sure what this is saying. You may want to take a quick edit pass on your comments.

Reply:
will do, in a another PR